app/src/main/java/com/kickstarter/libs/FirebaseAnalyticsClient.kt

```diff
@@ -0,0 +1,24 @@
+package com.kickstarter.libs
+
+import android.content.SharedPreferences
+import android.os.Bundle
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.kickstarter.libs.models.OptimizelyFeature
+import com.kickstarter.ui.SharedPreferenceKey.CONSENT_MANAGEMENT_PREFERENCE
+
+open class FirebaseAnalyticsClient(
+    private var optimizely: ExperimentsClientType,
+    private var preference: SharedPreferences,
+    private val firebaseAnalytics: FirebaseAnalytics?,
+) {
```

Comment:
You could have created here an interface having two methods { track, isEnabled} in a facade way. And add to the Dagger graph, the dependency with the interface type.